cores/esp8266/Esp.cpp

```diff
@@ -109,6 +109,13 @@ extern "C" void esp_yield();
 
 void EspClass::deepSleep(uint64_t time_us, WakeMode mode)
 {
+    if (time_us > deepSleepMax()) // we need to prevent the esp8266 from not waking up from deepsleep
```

Comment:
Hate to dogpile here, but if you do find out that `deepSleepMax()` doesn't really work and need to derate by 5% or whatever, please use that # to compare and not `deepSleepMax()` in this `if` condition.    Otherwise someone will pass in, say, `deepSleepMax() - 1` which won't work but which won't be caught by this check.

Reply:
Would it also be wise to increasingly loop the whole function until users's `time_us` is reached ?